File: python-src/BubbleSort.py
# Implementation of Bubblesort algorithm for sorting
class BubbleSort:
    
    # Sort input array
    def sort(A):
        
        # Swap array (A) elements at indices u and v
        def swap(A, u, v):
            temp = A[u]
            A[u] = A[v]
            A[v] = temp
            return
        
        a_len = len(A)

        # Compare adjacent list elements and exchange the
        # values if they are out of order.
        for i in range(0, a_len-1):
            for j in range(0, a_len-1-i):
                if (A[j+1] < A[j]):
                    swap(A, j, j+1)

        # Return the sorted list
        return A

File: java-src/BubbleSort.java
// Bubblesort algorithm
public class BubbleSort
{
    private int[] A;
    
    // Constructor
    public BubbleSort(int[] data)
    {
        A = data;
    }
    
    public int[] sort()
    {
        int i, j;
        
        // Compare adjacent elements of the array and exchange the 
        // values if they are out of order.
        for(i=0; i<A.length-1; i++)
        {
            for(j=0; j<A.length-1-i; j++)
            {
                if(A[j+1] < A[j])
                {
                    swap(A, j, j+1);
                }
            }
        }
        
        return A;
    }
    
    // Swap array elements at A[u] and A[v].
    private void swap(int[] A, int u, int v)
    {
        int temp;
        
        temp = A[u];
        A[u] = A[v];
        A[v] = temp;
    }
}